src/k2/httpproxy/HTTPProxy.cpp

```diff
@@ -441,6 +450,7 @@ HTTPProxy::HTTPProxy() : _client(K23SIClientConfig()) {
 
 seastar::future<> HTTPProxy::gracefulStop() {
     std::vector<seastar::future<>> futs;
+    _expiryList.stop().wait();
```

Comment:
calling wait or get should not be common practice as these block.just push it to the futures: futs.push_back(_expiryList.stop())

Reply:
Fixed.